### PR TITLE
Introduce scroll progress dots

### PR DESCRIPTION
**Pull request by rzadp**

### CODE REVIEW

**Comment by rzadp** [comment]
## React

Screenshot 2019-11-05 at 15 30 04
Screenshot 2019-11-05 at 15 30 09
Screenshot 2019-11-05 at 15 30 12
Screenshot 2019-11-05 at 15 30 21

**Comment by rzadp** [comment]
## Wallet

Screenshot 2019-11-05 at 15 38 48
Screenshot 2019-11-05 at 15 38 56
Screenshot 2019-11-05 at 15 38 59
Screenshot 2019-11-05 at 15 39 02

File: universal-login-react/src/ui/commons/ScrollProgressDots.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+interface ScrollProgressDotsProps {
+  dotOpacities: Array<number>;
+  minimumOpacity?: number;
+  maximumOpacity?: number;
+}
+
+export const ScrollProgressDots = ({dotOpacities, minimumOpacity = 0.0, maximumOpacity = 1.0}: ScrollProgressDotsProps) => {
+  dotOpacities = dotOpacities.map((opacity) => opacity < minimumOpacity ? minimumOpacity : opacity);
+  dotOpacities = dotOpacities.map((opacity) => opacity > maximumOpacity ? maximumOpacity : opacity);
```
Comment (cehali) [inline universal-login-react/src/ui/commons/ScrollProgressDots.tsx]: Maybe one map, something like this (maybe less sophisticated but you know what is going on at glance)
```js
dotOpacities = dotOpacities.map((opacity) => {
    if (opacity < minimumOpacity) {
          return minimumOpacity
     } else if (opacity > maximumOpacity) {
          return maximumOpacity
     }
     return opacity
}
```

Comment (Marik-D) [reply universal-login-react/src/ui/commons/ScrollProgressDots.tsx]:
```
const clamp = (val, min, max) => Math.max(Math.min(val, max), min)

 const dots = dotOpacities.map((opacity, i) =>
    <div key={i} style={{opacity: clamp(opacity, min, max)}} className='scroll-progress-dot'/>,
  );

Comment (Marik-D) [reply universal-login-react/src/ui/commons/ScrollProgressDots.tsx]: And place `clamp` in some utils

Comment (rzadp) [reply universal-login-react/src/ui/commons/ScrollProgressDots.tsx]: In a follow-up PR

**Pull request merged**